lib/inferno/dsl/fhir_evaluation/rules/valuesets_demonstrate.rb

```diff
@@ -0,0 +1,207 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'json'
+require 'uri'
+require 'base64'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        # This rule evaluates if an IG defines a new valueset, the examples should
+        # demonstrate reasonable coverage of that valueset.
+        # Note this probably only makes sense for small valuesets
+        # such as status options, not something like disease codes from SNOMED.
+
+        # Algorithm:
+        # 1. Extract pairs of system and code from include in value sets in IG
+        # 2. If valueSet exists in include, retrieve the value sets from UMLS.
+        #    Extract pairs of system and code from the result.
+        # 3. For each pair of system and code, check if any resources in the IG have instance of them.
+        # 4. Count total number of existences.
+
+        class ValueSetsDemonstrate < Rule
+          attr_accessor :config
+
+          def check(context)
```

Comment:
This method seems to be doing way too much. It's building the list of things to check, performing the check, and reporting the results. A method doing this much will be very difficult to unit test in a comprehensive way.

Reply:
Reorganized the method

https://github.com/inferno-framework/inferno-core/pull/636/commits/b547277ddd50ac4e5e7f1bd264d6eb46e3c0bd01